### PR TITLE
Add moshi-jsonapi

### PR DESCRIPTION
**Pull request by braczkow**
Probably my last input :(

Added `moshi-jsonapi`, added `assertJ`

Works fine.

### CODE REVIEW

**Review by MDikkii — CHANGES_REQUESTED** [review]

**Review by MDikkii — APPROVED** [review]

**Review by krzdabrowski — APPROVED** [review]

File: user/src/main/java/com/example/user/auth/data/dto/SessionDto.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.example.user.auth.data.dto
+
+import com.squareup.moshi.Json
+import moe.banana.jsonapi2.JsonApi
+import moe.banana.jsonapi2.Resource
+
+@JsonApi(type = "session")
+data class SessionDto(
+    @field:Json(name = "access_token") val access_token: String = "",
```
Comment (Ant8) [inline user/src/main/java/com/example/user/auth/data/dto/SessionDto.kt:9]: nit, I'd change fields names to snake case

File: base/src/main/java/com/example/base/testutils/TestUtils.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import moe.banana.jsonapi2.Resource
+import moe.banana.jsonapi2.ResourceAdapterFactory
+
+object TestUtils {
+
+    fun jsonApiMoshi(vararg clazzes: Class<out Resource>): Moshi {
+        val builder = ResourceAdapterFactory
+            .builder()
+
+        clazzes.forEach {
```
Comment (Ant8) [inline base/src/main/java/com/example/base/testutils/TestUtils.kt:13]: huh, this together with `@Provide into set` is quite clever, although perhaps not that readable at first sight

File: user/src/test/java/com/example/user/auth/data/dto/AccountDtoTest.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import com.example.base.testutils.TestUtils
+import moe.banana.jsonapi2.Document
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+
+private const val REGISTER_RESPONSE =
+    """{"data":{"id":"34db2713-0664-48fb-9be4-9c8c0e850051","type":"account","attributes":{"email":"barrr6611@cz.cz"},"relationships":{"session":{"data":{"id":"04f40a15-40be-4579-b85b-85d572d2f785","type":"session"}},"user":{"data":{"id":"f46c6d10-4216-4bb1-abc2-876085fc90aa","type":"user"}}}},"included":[{"id":"f46c6d10-4216-4bb1-abc2-876085fc90aa","type":"user","attributes":{"first_name":null,"last_name":null}},{"id":"04f40a15-40be-4579-b85b-85d572d2f785","type":"session","attributes":{"access_token":"access-token","refresh_token":"refresh-token"},"relationships":{"account":{"data":null},"user":{"data":null}}}]}"""
+
+class AccountDtoTest {
```
Comment (Ant8) [inline user/src/test/java/com/example/user/auth/data/dto/AccountDtoTest.kt:11]: :open_mouth:

**Review by Ant8 — APPROVED** [review]
generally LGTM

File: user/src/main/java/com/example/user/auth/data/dto/SessionDto.kt (hunk 1 above)
Comment (MDikkii) [reply user/src/main/java/com/example/user/auth/data/dto/SessionDto.kt]: Done

**Pull request merged**